utils/nctl/sh/scenarios/common/health_checks.sh

```diff
@@ -0,0 +1,304 @@
+#!/usr/bin/env bash
+
+source "$NCTL"/sh/scenarios/common/itst.sh
+
+# Exit if any of the commands fail.
+set -e
+
+function main() {
+    log "------------------------------------------------------------"
+    log "Starting health checks..."
+    log "------------------------------------------------------------"
+
+    log "... Stopping nodes to freeze logs"
+    nctl-stop
+
+    # Check allowed vs found counts
+    assert_error_count
+    assert_equivocator_count
+    assert_restart_count
+    assert_crash_count
+    assert_eject_count
+    assert_doppel_count
+
+    log "------------------------------------------------------------"
+    log "Health checks complete"
+    log "------------------------------------------------------------"
+}
+
+##########################################################
+# Function: assert_error_count
+# Uses: Variable ERRORS_ALLOWED
+# Description: Searches the log files counting instances
+#              of errors. Compares count against
+#              the total allowed by the test. If
+#              ERRORS_ALLOWED is set to ignore, it will
+#              return 0.
+#
+##########################################################
+
+function assert_error_count() {
+    local COUNT
+    local TOTAL
+
+    if [ "$ERRORS_ALLOWED" = "ignore" ]; then
+        log_step "Error count ignored by test, continuing..."
+        return 0
+    fi
```

Comment:
Instead of ignoring right away, maybe we can count the # of occurences and then, at the end, either compare it with the expected value or, in the case of `ignored`, log something along the lines of "Ignoring 12 errors". Similarly, for other healtchecks ("Ignoring 3 equivocations", etc.).

Reply:
https://github.com/casper-network/casper-node/pull/2020/commits/2066ce1fd55d1178cf34875ec75acd3009c4843c